dotnet/src/Connectors/Connectors.Google/Core/ClientBase.cs

```diff
@@ -109,4 +109,12 @@ protected void Log(LogLevel logLevel, string? message, params object[] args)
 #pragma warning restore CA2254
         }
     }
+
+    protected static string GetApiVersionSubLink(GoogleApiVersion apiVersion)
+        => apiVersion switch
+        {
+            GoogleApiVersion.Stable => "v1",
+            GoogleApiVersion.Beta => "v1beta",
+            _ => throw new NotSupportedException($"Google API version {apiVersion} is not supported.")
+        };
```

Comment:
The advantage of this approach is we are being strict about the versions we support. 

There are some disadvantages e.g.

- Stable may be v1 now but this will change over time but what happens if there are multiple stable versions supported. Developers should be able to select the exact version they want.
- If `v1.1beta` is released developers cannot switch to it to test if their existing functionality still works es expected with the newer release, they will need to wait for a new release of SK.

Elsewhere we are allowing the API version to be passed in as a string value which we insert into the URL. This provides control to the developer to try out new versions as soon as they are available. @RogerBarreto Thoughts?

Reply:
I can change this to string but... if we decide to move implementation to use google nuget, there is no place to specify api version. There are just two different nugets, package stable https://cloud.google.com/vertex-ai/docs/reference/rpc/google.cloud.aiplatform.v1 and package beta https://cloud.google.com/vertex-ai/docs/reference/rpc/google.cloud.aiplatform.v1beta1 (not available nuget for vertex currently)

Reply:
Follow the approach from Azure SDK, enum for each version:

[Check here](https://github.com/Azure/azure-sdk-for-net/blob/d12489ab27079466308dbd9dcc89879be7f02fe7/sdk/openai/Azure.AI.OpenAI/src/Generated/OpenAIClientOptions.cs#L14)

Reply:
@Krzysztof318 except for the approach how to enumerate them, which avoids @markwallace-microsoft concerns as well, we work with Latest and Specific enums.

Reply:
I changed to V1 and V1Beta but I didn't add Latest property because it doesn't make sense there. VertexAI gemini works now only with V1, not beta, but GoogleAI works only with Beta now.

Reply:
Should be different enumerators for each Platform, so you can have a latest for GoogleAI and a latest for VertexAI side by side.

Latest is not super necessary ATM, but thats the main idea, since those versions are related to different API's.

Reply:
yeah, I changed that, I don't want to introduce latest due to vertex didn't work correctly with Beta now in my tests.